debug/main.py

```diff
@@ -35,89 +35,83 @@ def bgp():
     @bgp.command('allow-martians')
     def allow_martians():
         """BGP allow martian next hops"""
-        command = 'sudo vtysh -c "debug bgp allow-martians"'
+        command = ['sudo', 'vtysh', '-c', "debug bgp allow-martians"]
         run_command(command)
 
     @bgp.command()
     @click.argument('additional', type=click.Choice(['segment']), required=False)
     def as4(additional):
         """BGP AS4 actions"""
-        command = 'sudo vtysh -c "debug bgp as4'
+        command = ['sudo', 'vtysh', '-c', "debug bgp as4"]
         if additional is not None:
-            command += " segment"
-        command += '"'
+            command = ['sudo', 'vtysh', '-c', "debug bgp as4 segment"]
         run_command(command)
 
     @bgp.command()
     @click.argument('prefix', required=True)
     def bestpath(prefix):
         """BGP bestpath"""
-        command = 'sudo vtysh -c "debug bgp bestpath %s"' % prefix
+        command = ['sudo', 'vtysh', '-c', "debug bgp bestpath %s" % prefix]
         run_command(command)
 
     @bgp.command()
     @click.argument('prefix_or_iface', required=False)
     def keepalives(prefix_or_iface):
         """BGP Neighbor Keepalives"""
-        command = 'sudo vtysh -c "debug bgp keepalives'
+        command = ['sudo', 'vtysh', '-c', "debug bgp keepalives"]
         if prefix_or_iface is not None:
-            command += " " + prefix_or_iface
-        command += '"'
+            command = ['sudo', 'vtysh', '-c', "debug bgp keepalives " + prefix_or_iface]
```

Comment:
>['sudo', 'vtysh', '-c', "debug bgp keepalives " [](http://example.com/codeflow?start=22&length=47)

The sub array repeats in code. Better to define an variable.

Reply:
Sorry for misleading. I am talking about specific issue that `'sudo', 'vtysh', '-c', "debug bgp keepalives "` is repeated twice. Not to mean a general issue.

However I see you take effort to improve the whole file, and it's better than what I was asking for.

Still this `vtysh_cmd + ["debug bgp keepalives "` repeated twice.